aoc/src/main/scala/Day03.scala

```diff
@@ -0,0 +1,69 @@
+import scala.io._
+
+object Day03 extends App:
+
+  val day: String =
+    this.getClass.getName.drop(3).init
+
+  val start1: Long =
+    System.currentTimeMillis
+
+  val input: List[String] =
+    Source
+      .fromResource(s"input$day.txt")
+      .getLines
+      .toList
+
+  def parser(s: String): (String,String) =
+    s.splitAt(s.length / 2)
+
+  def commonLetter(compartments: (String, String)): List[Char] =
+    val letters: List[String] = compartments._1.sliding(1).toList
+    for (
+      x <- letters;
+      y <- x
+      if compartments._2.contains(y)
+    ) yield y
+
+  def priority(list: List[Char]): List[Int] =
+    val alphabetLower = "abcdefghijklmnopqrstuvwxyz".zip(Stream.from(1)).toList
```

Comment:
this works - and is _very_ readable - but.. you could als use the fact that `Char` is numeric and that the character ranges `a-z` and `A-Z` increment - bc of this the number of a specific char can also be calculated with the one-liner `if c.isLower then c.toInt - 96 else c.toInt - 38`, ie. `a-z` transposed `-96` and `A-Z` transposed `-38`